packages/zwave-js/src/lib/commandclass/HumidityControlSetpointCC.ts

```diff
@@ -0,0 +1,939 @@
+		propertyKey: setpointType,
+	};
+}
+
+function getSetpointScaleValueID(
+	endpoint: number,
+	setpointType: number,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Humidity Control Setpoint"],
+		endpoint,
+		property: "setpointScale",
+		propertyKey: setpointType,
+	};
+}
+
+@API(CommandClasses["Humidity Control Setpoint"])
+export class HumidityControlSetpointCCAPI extends CCAPI {
+	public supportsCommand(
+		cmd: HumidityControlSetpointCommand,
+	): Maybe<boolean> {
+		switch (cmd) {
+			case HumidityControlSetpointCommand.Get:
+			case HumidityControlSetpointCommand.SupportedGet:
+				return this.isSinglecast();
+			case HumidityControlSetpointCommand.Set:
+				return true; // This is mandatory
+			case HumidityControlSetpointCommand.CapabilitiesGet:
+				return this.isSinglecast();
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	): Promise<void> => {
+		if (property !== "setpoint") {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+		if (typeof propertyKey !== "number") {
+			throw new ZWaveError(
+				`${
+					CommandClasses[this.ccId]
+				}: "${property}" must be further specified by a numeric property key`,
+				ZWaveErrorCodes.Argument_Invalid,
+			);
+		}
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		}
+
+		const preferredScale = this.endpoint
+			.getNodeUnsafe()!
+			.getValue<number>(
+				getSetpointScaleValueID(this.endpoint.index, propertyKey),
+			);
+		await this.set(propertyKey, value, preferredScale ?? 0);
+
+		if (this.isSinglecast()) {
+			// Verify the current value after a delay
+			this.schedulePoll({ property, propertyKey });
+		}
+	};
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+		propertyKey,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "setpoint":
+				if (typeof propertyKey !== "number") {
+					throw new ZWaveError(
+						`${
+							CommandClasses[this.ccId]
+						}: "${property}" must be further specified by a numeric property key`,
+						ZWaveErrorCodes.Argument_Invalid,
+					);
+				}
+
+				return (await this.get(propertyKey))?.value;
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+	public async get(setpointType: HumidityControlSetpointType) {
+		this.assertSupportsCommand(
+			HumidityControlSetpointCommand,
+			HumidityControlSetpointCommand.Get,
+		);
+
+		const cc = new HumidityControlSetpointCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			setpointType,
+		});
+		const response =
+			await this.driver.sendCommand<HumidityControlSetpointCCReport>(
+				cc,
+				this.commandOptions,
+			);
+		if (!response) return;
+		return response.type === HumidityControlSetpointType["N/A"]
+			? // not supported
+			  undefined
```

Comment:
Is this setpoint type actually returned? If so, it could use a comment explaining it. The specs don't seem to mention it.

Reply:
For the same reason `N/A` is set in the enum at index 0... There's no type 0 defined in the spec so if the command returns a response with type 0 for any reason, we respond with undefined... Copied from the Thermostat Setpoint CC which works the same way https://github.com/zwave-js/node-zwave-js/blob/583645c6618ae1abb2666eb81806f8d73c97d719/packages/zwave-js/src/lib/commandclass/ThermostatSetpointCC.ts#L233-L241

![Screenshot from 2022-01-09 17-18-37](https://user-images.githubusercontent.com/2053039/148703426-97df7cb3-cadc-4005-80f1-5afd4618e6f7.png)

Reply:
AFAIK the Thermostat CCs actually define a setpoint type N/A, but I'm okay with this.